src/service/api.tsx

```diff
@@ -127,11 +131,52 @@ const api = {
   },
 
   async getSearchList(query: string, page: number = 1) {
-    let obj = await axios.get(`search/movie?${API_KEY}&query=${query}&page=${page}`);
+    const obj = await axios.get(`search/movie?${API_KEY}&query=${query}&page=${page}`);
     return obj.data.results;
   },
 
-  changeImgLinks(url: string, size: string = 'w500') {
+  async getSearchFileredList(dataFilter:
+    {
+      from: number,
+      to: number,
+      genre: string | null,
+      page: number
+    } = {
+      from: 1900,
+      to: 2020,
+      genre: null,
+      page: 1,
+    }
+  ) {
+
+    function setFilteredData() {
+      const { from, to } = dataFilter
+      if (from !== to) return `&primary_release_date.gte=${from}-01-01&primary_release_date.lte=${to}-01-01`
+      return `&primary_release_date.gte=${from}-01-01`
+    }
+
+    function setGenre() {
+      const { genre } = dataFilter;
+      if (!genre) {
+        const genreId = API_GENRE_ID.filter((item: any) => Object.keys(item)[0] === 'comedy');
+        return `&with_genres=${Object.keys(genreId[0])[0]}`
```

Comment:
do you want to set default to comedy? 
API_GENRE_ID[Genres.comedy]
https://github.com/oleksandrstarov/sigma_js_internship_2021/blob/487042caecb3276120fe3aa1b394ca6b3d631c45/src/components/SearchField.tsx#L8